### DISCUSSION TITLE
composition relationship

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [prisma/prisma#21082] [Q&A] [answered]
### Question

I am developing a clone social media api for the post their will be many comment and there is no comment without a post so the post is composed with comment how can I implement this relationship in schema.prisma

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @Elmatbaisaad 👋 

Thank you for raising this question.

To model a relationship between `Posts` and `Comments` where each post can have multiple comments and there are no comments without a post, your schema will look like:
```prisma
model Post {
  id       Int       @id @default(autoincrement())
  title    String
  content  String
  comments Comment[] // Define a one-to-many relationship to comments
}

model Comment {
  id     Int      @id @default(autoincrement())
  text   String
  postId Int      // Define a foreign key to the Post model
  post   Post     @relation(fields: [postId], references: [id]) // Define the reverse relationship to Post
}
```

With this schema, you can easily create, retrieve, and query posts and their associated comments using Prisma. Each comment is linked to a specific post, ensuring that there are no comments without a post.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by Elmatbaisaad** [comment]
Thank you for your answer it helped me and it answer perfectly my question

**Reply by ludralph** [reply]
You're welcome 🙌
